src/pages/index.js

```diff
@@ -0,0 +1,335 @@
+  ".modal__close_type_preview"
+);
+
+//card Elements
+const cardTemplate = document.querySelector("#card-template");
+const cardList = document.querySelector(".cards__list");
+
+//imported Images
+const headerImg = document.getElementById("header-logo");
+headerImg.src = headerLgo;
+const avatarImg = document.getElementById("avatar-img");
+avatarImg.src = avatarPic;
+const pencilImg = document.getElementById("pencil-icon");
+pencilImg.src = pencilPic;
+const plusIcon = document.getElementById("plus-icon");
+plusIcon.src = plusPic;
+
+function handleEscape(evt) {
+  if (evt.key === "Escape") {
+    const openModal = document.querySelector(".modal_is-opened");
+    if (openModal) {
+      closeModal(openModal);
+    }
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_is-opened");
+
+  document.addEventListener("keyup", handleKeydown);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_is-opened");
+
+  document.removeEventListener("keyup", handleKeydown);
+}
+
+function handleEditProfileSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  console.log(submitBtn);
+
+  api
+    .editUserInfo({
+      name: nameInput.value,
+      about: descriptionInput.value,
+    })
+    .then((data) => {
+      profileNameEl.textContent = data.name;
+      profileDescriptionEl.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleCardSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  console.log(submitBtn);
+
+  const name = captionInput.value;
+  const link = linkInput.value;
+
+  api
+    .addCard({ name, link })
+    .then((res) => {
+      const cardEl = getCardElement(res);
+      cardList.prepend(cardEl);
+      closeModal(cardModal);
+      cardForm.reset();
+      disableButton(cardSubmitBtn, settings);
+    })
+    .catch((err) => console.error("Failed to add card:", err))
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  console.log(submitBtn);
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      avatarImg.src = data.avatar;
+      closeModal(avatarModal);
+      avatarForm.reset();
+      disableButton(avatarSubmitBtn, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  if (!cardId) {
+    console.error("Card ID is missing!");
+    return;
+  }
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  api
```

Comment:
According to the task and design, you should change the button text to `Deleting...` while waiting for the server response when you delete a card